hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/master/BackupLogCleaner.java

```diff
@@ -80,8 +81,13 @@ public void init(Map<String, Object> params) {
     }
   }
 
-  private Map<Address, Long> getServersToOldestBackupMapping(List<BackupInfo> backups)
-    throws IOException {
+  private Map<Address, Long> getServerToLastBackupTs(List<BackupInfo> backups) throws IOException {
```

Comment:
imho this method name is still confusing. What's the relationship between "last" and "oldest"? Can backups not be chronologically ordered? How about `getServerToOldestBackupTS`, and the log message can talk about cleaning WALs that are older than the **oldest** backups.

Reply:
"Last" could indeed be interpreted as "most recent". I also prefer "Oldest" (or "Earliest")

Reply:
Good suggestions, ty both

Reply:
Oh, to clarify, we don't want to clean WALs that are older than the oldest backup. We want to clean WALs that are older than the newest backup. It's confusing because this was literally called the opposite, `getServersToOldestBackupMapping`, but I believe newest backup makes sense and the implementation clearly fetches the newest, despite what the name suggests: https://github.com/apache/hbase/blob/master/hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/master/BackupLogCleaner.java#L83-L102

```java
  private Map<Address, Long> getServersToOldestBackupMapping(List<BackupInfo> backups)
    throws IOException {
    Map<Address, Long> serverAddressToLastBackupMap = new HashMap<>();

    Map<TableName, Long> tableNameBackupInfoMap = new HashMap<>();
    for (BackupInfo backupInfo : backups) {
      for (TableName table : backupInfo.getTables()) {
        tableNameBackupInfoMap.putIfAbsent(table, backupInfo.getStartTs());
        if (tableNameBackupInfoMap.get(table) <= backupInfo.getStartTs()) {
          // ie, if backup is *newer* than what we've already mapped, then overwrite
          tableNameBackupInfoMap.put(table, backupInfo.getStartTs());
          for (Map.Entry<String, Long> entry : backupInfo.getTableSetTimestampMap().get(table)
            .entrySet()) {
            serverAddressToLastBackupMap.put(Address.fromString(entry.getKey()), entry.getValue());
          }
        }
      }
    }

    return serverAddressToLastBackupMap;
  }
```

Reply:
I'm pro naming this `newest`.

Realizing now that this tracks the newest backup, it means this logic also contains a bug: it does not keep track of multi-root backups. I.e. if there are 2 backups roots R1 and R2, where R1 was backed up 1 week ago, and R2 today, the WALS needed for R1 could be deleted (= data loss).
Perhaps log that as a separate issue?

Reply:
It's also a good idea to use "newest" or "most recent" in the logs produced in this class, rather than "latest" or "last".

Reply:
> Realizing now that this tracks the newest backup, it means this logic also contains a bug: it does not keep track of multi-root backups. I.e. if there are 2 backups roots R1 and R2, where R1 was backed up 1 week ago, and R2 today, the WALS needed for R1 could be deleted (= data loss).
Perhaps log that as a separate issue?

Agreed this sounds like a separate, but very real, issue. This multi-root case seems to be a gift that keeps giving...

Reply:
Hadn't seen an issue for this passing by yet, so logged it as https://issues.apache.org/jira/browse/HBASE-28833